muikku-core-plugins/src/main/java/fi/muikku/plugins/workspace/WorkspaceJournalBackingBean.java

```diff
@@ -130,19 +163,66 @@ public Long getWorkspaceJournalEntryId() {
   public void setWorkspaceJournalEntryId(Long workspaceJournalEntryId) {
     this.workspaceJournalEntryId = workspaceJournalEntryId;
   }
-
-  private Long workspaceEntityId;
   
+  public void setStudentId(Long studentId) {
+    this.studentId = studentId;
+  }
+  
+  public Long getStudentId() {
+    return studentId;
+  }
+  
+  public boolean isMyJournal() {
+    return studentId == null;
+  }
+  
+  public boolean isCanListAllEntries() {
+    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
+    return sessionController.hasCoursePermission(
+        MuikkuPermissions.LIST_ALL_JOURNAL_ENTRIES, workspaceEntity);
+  }
+  
+  public List<UserWithUserEntity> getWorkspaceStudents() {
+    ArrayList<UserWithUserEntity> result = new ArrayList<>();
```

Comment:
Initialize in PostConstruct and just return it here, this solution can lead to performance issues